src/footer/script1.js

```diff
@@ -0,0 +1,222 @@
+            us_cart.forEach(async function (el) {
+                el.insertAdjacentHTML('afterend', '<hr><div id="upsell-container"></div>');
+            });
+        }
+    } else {
+        us_main_div.forEach(function (el) {
+            el.innerHTML = '';
+        });
+    }
+}
+
+async function cacheResults(result) {
+    result.recommendations.forEach(async function (recommendation) {
+        recommendation = await getVariantDetail(recommendation);
+        let recommendationJson = JSON.stringify(recommendation);
+        if (recommendationJson != null) {
+            sessionStorage.setItem('us_' + recommendation.id, recommendationJson.toString());
+        }
+    });
+    return true;
+}
+
+async function getVariantDetail(recommendation) {
+    return await getVariantDetailFromEshop(recommendation);
+}
+
+async function getVariantDetailFromEshop(recommendation)
+{
+    const us_link_response = await goToURL(insertCacheInUrl(recommendation.url));
+    const response = await us_link_response.text();
+    const tempContainer = document.createElement('div');
+    tempContainer.innerHTML = response;
+    const availability = tempContainer.querySelector('span.availability-label');
+    recommendation.availability = '-';
+    if (availability) {
+        recommendation.availability = availability.textContent.trim();
+    }
+    const product_id = tempContainer.querySelector('input[name="productId"]');
+    if (product_id) {
+        recommendation.id = product_id.getAttribute('value');
+    } else {
+        return null;
+    }
+    const price_id = tempContainer.querySelector('input[name="priceId"]');
+    if (price_id) {
+        recommendation.priceId = price_id.getAttribute('value');
+    } else {
+        return null;
+    }
+    recommendation.color = null;
+    if (availability.style.color) {
+        recommendation.color = availability.style.color;
+    }
+    const price = tempContainer.querySelector('span.price-final-holder')
+    if (price) {
+        recommendation.price = price.textContent.trim();
+    }
+    if (recommendation.guid)
+    return recommendation;
+}
+
+async function goToURL(url) {
+    try {
+        const response = await fetch(url);
+        return response;
+    }
+    catch (error) {
+        console.error('Error:', error);
+    }
+}
+
+async function cacheRequest(us_cart_items, result) {
+    let us_recommendation_ids = [];
+    result.recommendations.forEach(element => {
+        us_recommendation_ids.push(element.id);
+    });
+    sessionStorage.setItem('us_request_' + us_cart_items.toString(), us_recommendation_ids.toString());
+    sessionStorage.setItem('us_header', result.header);
+    return true;
+}
+
+function checkCachedData(us_product_ids) {
+    let us_result = true;
+    us_product_ids.forEach(function (product_id) {
+        if (sessionStorage.getItem('us_' + product_id) == null) {
+            us_result = false;
+        }
+    });
+    return us_result;
+}
+
+async function printResults() {
```

Comment:
Nemohla by funkce být synchronní? Nepracuje s žádnými blokujícím zdroji.
```suggestion
function printResults() {
```

Reply:
Funkce je nyní synchronní, děkuji